### PR TITLE
A restructure/removal/cleanup of Ninja Code [Part 1]

### PR DESCRIPTION
**Pull request by RemieRichards**
- RP Objective removed, Admins can manually assign these, having one auto generated with a risk of no admins about is silly
- Kill Deathsquad and Kill Xeno Queen objectives removed, these are such rare cases plus the Ninja will probably do these anyways
- Kamikaze mode is removed (One: OP, Two: From my research impossible to obtain without admins <B>Edit:</B> Apparently it's possible to obtain 100% of the time, still, not staying.)
- Suit AI Removed
- Suit pAI Removed
- ninja.dm split up into it's own /module folder structure /modules/ninja
- Removes energyblade power (item still exists for badmins, but it's psuedo-emag functionality is gone)
- Replaces energy blade with a real physical energy katana (slightly buffed +10, since you can lose it)
- Energy Katana can emag any object it hits, like the energyblade used to hackily try and do
- Marked every external source of ninja code deemed "snowflakey" with the tag `//NINJACODE` (Explained in readme)(mainly for me in a Part 2/3 PR)
- Head items now have a `blockTracking` var to fix ninja helmets snowflakely not being seen by AIs

And a bunch of other misc fixes such as colon removals.
A few powers were cleaned, but besides the above points it's mainly a restructure.

I intend to go over the powers on a case by case basis in a "Part 2"

### CODE REVIEW

**Comment by WJohn** [comment]
So uh

Is it really happening this time?

Actually happening?

**Comment by hornygranny** [comment]
probably

**Comment by phil235** [comment]
:japanese_ogre:

**Comment by Kromgar** [comment]
Can the ninja reclaim the Katana with a power? Like he has it fly to his hands or some shit? Has to be on the screen and the person he's taking it from nearby and it could just blast through windows and doors

**Comment by RemieRichards** [comment]
@Kromgar tough ask(not the code, this coming next): but if you remind me, in the Part 2 PR (the one I'll focus mainly on powers), I will indeed add that, since it's pretty cool.

**Comment by Dorsisdwarf** [comment]
When you have the part 2 PR give ninjas the cloak of darkness mutation so they're hard to see in the dark

**Comment by Atskadan** [comment]
>42 file changes

File: code/modules/ninja/energy_katana.dm [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+
+/obj/item/weapon/katana/energy
+	name = "energy katana"
+	desc = "a katana infused with a strong energy"
+	icon_state = "energy_katana"
+	item_state = "energy_katana"
+	force = 40
+	throwforce = 20
+	slot_flags = SLOT_BELT
```
Comment (Atskadan) [inline code/modules/ninja/energy_katana.dm]: add backslot too pls
everyone knows true space ninjas have them on the back

Comment (RemieRichards) [inline code/modules/ninja/energy_katana.dm]: The definition of ```slot_flags = SLOT_BELT``` was to get rid of the back slot, but sure I guess, I'll add it back. (I disagreed with it for an ENERGY katana, but sure).

**Comment by ike709** [comment]
RR for best coder

**Comment by Lumipharon** [comment]
Neat. Everything looks good except two things.
1)Pleb ninjas losing their RAEP katana, they need something like the recall spell or it's just goin to be embarrasing.
2) Why remove the RP objective? It's RP, even if they don't get muh green texts, it's a pretty fun thing if ninja's decide to actually try achieve, and if they choose to ignore it they're just turbomurderboner nerds.

Actually, policy was updated to make antags not go AGAINST objectives, which was in part to stop ninjas murderboning when their rp objective was shit like 'minimise casualties'.

**Comment by RemieRichards** [comment]
@Lumipharon 
Point 1: Remind me for PR 2 (The power oriented one, I've already agreed to do the sword recall for that)
Point 2: It's forced, admins have the power to add RP objectives, the game really shouldn't add potentially uncompletable objectives. (I won't budge on this one unless requested to)

**Comment by PKPenguin321** [comment]
you can get kamikaze mode without admins and it's actually fairly simple, not really OP since it kinda gibs you
idk remove it if you have to but it's a fun enough feature for the people who know about it

**Comment by RemieRichards** [comment]
@PKPenguin321 yes I've been let in on the secret to getting it 100% of the time, but It's still going, It's OP but it's also pointless since it kills you off like mad.

**Comment by palpatine213** [comment]
@RemieRichards If you disengage the suit, it stops killing you. With full health and a big enough battery, you can last for a decent amount of time.

**Comment by dannno** [comment]
i don't remember if random event ninja was removed but if it wasn't you should remove it

**Comment by dannno** [comment]
ps. whoever actually added ninja in the first place is a fucking retard, absolutely shameful

**Comment by Atskadan** [comment]
>i don't remember if random event ninja was removed but if it wasn't you should remove it

no
just make it less common tbh

**Comment by dannno** [comment]
having a random chance for this over-powered as shit antag only means that there is a chance for things already going on in the round to be fucked by this bullshit 

ninja was not thought about before implementing
an antag that can go invisible and is therefore untouchable by anything that isn't a suicide bomb is awful, especially since any decent player who upgrades their battery can crit 3-4 people easily, while fucking untouchable, and then go hide and recharge

i cannot fathom why anyone would want this shit to happen at random
it's literally a random chance to fuck the round up

**Comment by palpatine213** [comment]
Increasing the minimum time before it occurs would be a good idea tbh. 2 hours would be a good mark.

**Comment by Atskadan** [comment]
dannos just about every event can fuck over a round. that is literally the reason they exist. 
and remie is working right now on making ninjas not shit, so cool your jets

**Comment by dannno** [comment]
>dannos just about every event can fuck over a round

nothing can do it as easily as ninja
nothing
ninja is irredeemable

**Comment by Atskadan** [comment]
yes that is also the point of ninja

**Comment by palpatine213** [comment]
Danno, Admins can fuck over a round far more easily than a ninja can. Should they be removed?

**Comment by dannno** [comment]
that's false equivalency

**Comment by Atskadan** [comment]
yes it is actually but ninjas are still supposed to be OP

**Comment by dannno** [comment]
for what reason
it's simply not fun to be killed, at random, in the middle of a situation that has nothing to do with the ninja, with no way to see it coming or fight back
it's bad
why do you want this
i've seen it happen so many times, why do you want this

**Comment by Lumipharon** [comment]
Danno.
This is why policy was changed, so antags cant go directly against objectives.
IE: Ninjas cannot kill their protect objective, kill pointlessly if they have a 'minimise casualties etc etc.

Ninja is genuinely fun if you try play it properly and do your objectives stealthily and not be a turbomurderboner. If they always had a minimise objective, then it would never be a problem.

**Comment by Atskadan** [comment]
>remie is working right now on making ninjas not shit, so cool your jets
>remie is working right now on making ninjas not shit, so cool your jets
>remie is working right now on making ninjas not shit, so cool your jets
>remie is working right now on making ninjas not shit, so cool your jets
>remie is working right now on making ninjas not shit, so cool your jets

**Comment by Jordie0608** [comment]
Guess what.

**Comment by CorruptComputer** [comment]
Changelog?

**Comment by Miauw62** [comment]
remie richards more like remie BASEDchards amirite

**Comment by Miauw62** [comment]
seriously though this is great

**Comment by Atskadan** [comment]
that doesnt even sound similar miauw

**Comment by RemieRichards** [comment]
I appreciate the sentiments but yes, that sounds silly.

**Comment by Miauw62** [comment]
thats why i added amirite, learn 2 dank memes

**Comment by optimumtact** [comment]
Since you're doing all the impossible things

bring back the tensioner plox

**Comment by RemieRichards** [comment]
hasn't incoming sort of done that with his mulligan rounds? all a tensioner would be now a days would be an infinite series of mulligans, surely?

**Comment by RemieRichards** [comment]
It Begins.

**Comment by Jordie0608** [comment]
=(

**Pull request merged**